lib/web_git/diff.rb

```diff
@@ -0,0 +1,249 @@
+  module WebGit
+    class Diff
+      def self.get_diff
+        Dir.chdir(Rails.root) do
+          `git diff`
+        end
+      end
+
+      def self.get_each_left(diff)
+        filenames = self.get_file_names("")
+        files = self.get_most_of_each_file(diff, filenames)
+        lefts = {}
+        files.each_with_index do |file, i|
+          file_content = ""
+          lines = file.split("\n").drop(4)
+          start_line = 0
+          current_line_index = 0
+          line_number = start_line + current_line_index
+          lines.each do |line|
+            if !line.match?(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/)
+              if line.first != "+"
+                file_content += "#{line_number}| " + line + "\n"
+                line_number += 1
+              end
+            else
+              current_line_index = 0
+              # The line numbers in the output of a git diff match this regex
+              numbers = line.scan(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/).map(&:join)
+              # If left, starting line number is the first one in a split Array
+              start_line = numbers.first.split(" ").first.
+                split(",").first.to_i.abs
+              line_number = start_line + current_line_index
+              file_content += "\n"
+            end
+          end
+          lefts[filenames[i]] = file_content.chomp "\n"
+        end
+        lefts
+      end
+
+      def self.get_each_right(diff)
+        filenames = get_file_names("")
+        files = get_most_of_each_file(diff, filenames)
+        rights = {}
+        files.each_with_index do |file, i|
+          file_content = ""
+          lines = file.split("\n").drop(4)
+          start_line = 0
+          current_line_index = 0
+          line_number = start_line + current_line_index
+          lines.each do |line|
+            # The line numbers in the output of a git diff match this regex
+            # @@ -61,18 +61,15 @@
+            if !line.match?(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/)
+              if line.first != "-"
+                file_content += "#{line_number}| " + line + "\n"
+                line_number += 1
+              end
+            else
+              current_line_index = 0
+              numbers = line.scan(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/).map(&:join)
+              # If right, start line is the second in a split Array
+              start_line = numbers.first.split(" ").second.
+                split(",").first.to_i.abs
+              line_number = start_line + current_line_index
+              file_content += "\n"
+            end
+          end
+          rights[filenames[i]] = file_content.chomp "\n"
+        end
+        rights
+      end
+
+      def self.get_last_commit_hash
+        Dir.chdir(Rails.root) do
+          log = `git log -1 --oneline`
+          log.split.first
+        end
+      end
+
+      def self.get_most_of_each_file(diff, filenames)
```

Comment:
If you'd like to simplify this method, I tested this out and it seemed to result in the same return as your method. You could rewrite the `self.get_most_of_each_file` method to:

```ruby
def self.get_most_of_each_file(diff)
  diff.scan(/diff --git.*?(?=diff --git|\z)/m)
end
```
This would also eliminate the need for the `match_other_files` method.
Maybe renaming `get_most_of_each_file` to `file_diffs` might be more descriptive? That's not perfect but I think finding something that communicates the method returns a collection of diffs separated by file would be nice.

If you wanted to refactor it further, I might consider eliminating the argument and instead call `get_diff` inside the definition. So it might look like: 
```ruby
def self.file_diffs
  get_diff.scan(/diff --git.*?(?=diff --git|\z)/m)
end
```